lib/rrule.js

```diff
@@ -2209,13 +2209,41 @@
       }
     },
 
+    /**
+     * Filters out floating timezone properties.
+     * See https://tools.ietf.org/html/rfc5545#section-3.2.19 for more on the time zone property part of the iCal spec.
+     * See https://tools.ietf.org/html/rfc5545#section-3.3.5 for more on floating times in the iCal spec.
+     * @param {String} stringRule A string containing the set of recurrence properties.
+     * @return {String}           A filtered string containing the set of recurrence properties without floating
+     *                            timezones.
+     */
+    filterFloatingTimezones: function (stringRule) {
+      // Uses a regex to split the recurrence string by carriage return.
+      var newline = /[\n\r]/;
+      var parsedRecurrenceString = stringRule.split(newline)
+
+      // Remove any of the recurrence date properties that contain floating timezone properties.
+      parsedRecurrenceString.slice(0).forEach(function (ruleString) {
+        var rruleIndex = parsedRecurrenceString.indexOf(ruleString)
+
+        if (!contains(ruleString, 'RRULE') && rruleIndex !== -1) {
```

Comment:
Yeah it's just kinda not so great that you have to essentially modify the submodels and create your own client side model, to prevent having to worry about carrying state, just because the library doesn't support on part of the spec